client/components/tinymce/plugins/mentions/suggestion.jsx

```diff
@@ -13,28 +13,24 @@ const getRegExpFor = function( type, textToHighlight ) {
 };
 
 const highlight = function( content, textToHighlight, type ) {
-	const matcher = getRegExpFor( type, textToHighlight ),
-		matches = matcher.exec( content );
+	const matcher = getRegExpFor( type, textToHighlight );
+	const matches = matcher.exec( content );
 
-	if ( matches ) {
-		const highlights = [];
-		let highlighted = false;
-
-		for ( let i = 1, length = matches.length; i < length; i++ ) {
-			let item = matches[ i ];
-
-			if ( textToHighlight.toLowerCase() === item.toLowerCase() && ! highlighted ) {
-				item = <strong className="mentions__highlight" key={ i }>{ matches[ i ] }</strong>;
-				highlighted = true;
-			}
+	if ( ! matches ) {
+		return [ content ];
+	}
 
-			highlights.push( item );
+	return matches.map( ( item, i ) => {
+		if ( i === 0 ) {
+			return '';
 		}
 
-		return highlights;
-	}
+		if ( textToHighlight.toLowerCase() === item.toLowerCase() ) {
+			return <mark className="mentions__highlight" key={ i }>{ item }</mark>;
```

Comment:
it's best to use something other than the list index as a key here because the key is used to resolve differences in the DOM layout. it could be much better to use something like `item` as the key